moveit_ros/hybrid_planning/local_planner/local_planner_component/src/local_planner_component.cpp

```diff
@@ -235,17 +235,17 @@ void LocalPlannerComponent::executeIteration()
     // If the planner received an action request and a global solution it starts to plan locally
     case LocalPlannerState::LOCAL_PLANNING_ACTIVE:
     {
-      // Read current planning scene
       planning_scene_monitor_->updateSceneWithCurrentState();
-      planning_scene_monitor_->lockSceneRead();  // LOCK planning scene
-      planning_scene::PlanningScenePtr planning_scene = planning_scene_monitor_->getPlanningScene();
-      planning_scene_monitor_->unlockSceneRead();  // UNLOCK planning scene
 
-      // Get current state
-      auto current_robot_state = planning_scene->getCurrentStateNonConst();
+      // Read current robot state
+      moveit::core::RobotStatePtr current_robot_state;
+      {
+        planning_scene_monitor::LockedPlanningSceneRO ls(planning_scene_monitor_);
+        current_robot_state = std::make_shared<moveit::core::RobotState>(ls->getCurrentState());
+      }
```

Comment:
Is there a reason this needs to be a new shared printer and you couldn't just do this since you are making a copy:
```suggestion
      moveit::core::RobotState const current_robot_state = [&]()
      {
        planning_scene_monitor::LockedPlanningSceneRO ls(planning_scene_monitor_);
        return ls->getCurrentState();
      }
```

This is a bit of optimization but I do think it is good not to use less fancy language features (like shared_prt) when you don't need them.  Then later in the code, you don't need to add the dereference operator `*` to the calls to getTrajectoryProgress and getLocalTrajectory.

Reply:
You say "not to use less fancy features" and propose to use IIFE in the same comment. :rocket: :eyes:  :bouquet: 
I agree, though (1) `const` in second place in the type is very unusual for our coding style (I've never seen it anywhere), and (2) `[&](){` simplifies to `[&]{`.

It would be even better though to provide the method with that interface directly with the `PlanningSceneMonitor` class to begin with!

Reply:
I updated my suggestion.  IFEE is an advanced technique but it uses a much simpler and less complex language feature (lambdas) than shared_ptr which is basically adding a global and an extra call to malloc where it isn't needed.  I'm just here to continue my crusade against shared_ptr and for making more things const.  :shrug: :fire:

Reply:
On east const.  I also updated my suggestion to remove it.  I know we don't use that style in MoveIt.  It isn't a new thing even if you haven't seen it: https://www.youtube.com/watch?v=fv--IKZFVO8

Reply:
I'm OK with the lambda function. I don't think it's tooo complicated. Had to tweak your suggestion to capture _this_ instead of `planning_scene_monitor_`, though.

`error: capture of non-variable ‘moveit::hybrid_planning::LocalPlannerComponent::planning_scene_monitor_’`

Reply:
> I don't really care that much except that @v4hn's statement about not seeing east const before

I've seen it before, just not in MoveIt or even the ROS community. I'm absolutely ok with a clang-tidy option to use it everywhere if there are strong opinions among multiple maintainers. I don't think clang-tidy merged support for it yet though...
I also think we have more urgent issues and you are exaggerating syntactic details.
Let's continue this some time in a pub. :beers: